src/main/java/org/eclipse/yasson/internal/AnnotationFinder.java

```diff
@@ -0,0 +1,71 @@
+package org.eclipse.yasson.internal;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Finds an annotation including inherited annotations (e.g. meta-annotations).
+ * 
+ * @author JohT
+ */
+class AnnotationFinder<T extends Annotation> {
+
+    private final Class<T> annotationClass;
+
+    /**
+     * Gets the {@link AnnotationFinder} for the given Annotation-Type.
+     * 
+     * @param annotation {@link Class}, that is a sub-type of {@link Annotation}
+     * @return {@link AnnotationFinder}
+     */
+    public static final <S extends Annotation> AnnotationFinder<S> findAnnotation(Class<S> annotation) {
+	return new AnnotationFinder<>(annotation);
+    }
+
+    /**
+     * Only for testing purposes. <br>
+     * Please use a static factory method e.g. {@link #findAnnotation(Class)}.
+     * 
+     * @param annotation {@link Class}
+     */
+    protected AnnotationFinder(Class<T> annotation) {
+	this.annotationClass = annotation;
+    }
+
+    public T in(Annotation[] annotations) {
+	return findAnnotation(annotations, annotationClass, new HashSet<>());
+    }
+
+    /**
+     * Searches for annotation, collects processed, to avoid StackOverflow.
+     */
+    // This method is a copy of "AnnotationIntrospector.findAnnotation"
+    @SuppressWarnings("unchecked")
+    private T findAnnotation(Annotation[] declaredAnnotations, Class<T> annotationClass, Set<Annotation> processed) {
```

Comment:
Do we need to make a copy? What is the reason not to use AnnotationIntrospector?

Reply:
I'll make it static and move it instead of a copy.
I tried to get the changes into new classes, so it is easier to review, easier to unit test and less error prone (since i don't mess too much with the original code). Since i also need to find an annotation and i like the meta-annotation aware approach, i need a way to reuse it.